lib/evaluators/evaluatorHelpers/BaseExpressionEvaluator.ts

```diff
@@ -9,14 +10,14 @@ import type { SuperTypeCallback, TypeCache, ISuperTypeProvider } from '../../uti
 export interface ISharedContext {
   now?: Date;
   baseIRI?: string;
-  overloadCache?: OverLoadCache;
+  /**
+   * @deprecated Deprecated since version ... provided value will be ignored.
```

Comment:
Is that correct? Because I see you're still passing values to it?
But the values are just `{}`, which are not applicable for `LRUCache`.

Reply:
This should be correct, yes. So the `ISharedContext` is what the user will provide an engine. And an `ICompleteSharedContext` is what the engine will use. I kept the cache in `ICompleteSharedContext` because that way, each engine has it's own cache which is nice for testing.

I hope this answers your question?

Reply:
Ok, but maybe we should make it of type `any` then, so that we can remove the typings dependency on lru-cache.

Reply:
Seeing how we need the LRUCache anyway, do we keep the `LRUCache` type here or the `any` type?

Reply:
Sure, we can keep it then.